docs/source/distributed.tensor.parallel.rst

```diff
@@ -27,31 +27,20 @@ Tensor Parallelism supports the following parallel styles:
 .. autoclass:: torch.distributed.tensor.parallel.style.ColwiseParallel
   :members:
 
-.. autoclass:: torch.distributed.tensor.parallel.style.PairwiseParallel
-  :members:
-
-.. warning ::
-    Sequence Parallelism are still in experimental and no evaluation has been done.
-
-.. autoclass:: torch.distributed.tensor.parallel.style.SequenceParallel
-  :members:
-
 Since Tensor Parallelism is built on top of DTensor, we need to specify the
-input and output placement of the module with DTensors so it can expectedly
-interacts with the module before and after. The followings are functions
-used for input/output preparation:
+DTensor layout of the input and output of the module so it can interact with
+the module parameters and module afterwards. Users can achieve this by specifying
+the ``input_layouts`` and ``output_layouts`` which decide what the DTensor of inputs
+and outputs.
 
+If users only want to annotate the DTensor layout for inputs/outputs and no need to
+distribute its parameters, the following classes can be used in the ``parallelize_plan``
+of ``parallelize_module``:
 
-.. currentmodule:: torch.distributed.tensor.parallel.style
 
-.. autofunction::  make_input_replicate_1d
-.. autofunction::  make_input_reshard_replicate
-.. autofunction::  make_input_shard_1d
-.. autofunction::  make_input_shard_1d_last_dim
-.. autofunction::  make_output_replicate_1d
-.. autofunction::  make_output_reshard_tensor
-.. autofunction::  make_output_shard_1d
-.. autofunction::  make_output_tensor
+.. currentmodule:: torch.distributed.tensor.parallel.style
+.. autofunction::  PrepareModuleInput
+.. autofunction::  PrepareModuleOutput
 
 Currently, there are some constraints which makes it hard for the ``MultiheadAttention``
```

Comment:
Let's update this part of doc too:


Currently we recommend users to use ``ColwiseParallel``
and ``RowwiseParallel`` for each Linear/Embeeding layers to compose Attention Parallelization. There might be some code changes needed since we are parallelizing on the head dim of the ``MultiheadAttention`` module.

Reply:
sure.

Reply:
Maybe I should do this in the next PR.